test/text/ssa_text_parser_unit.js

```diff
@@ -0,0 +1,373 @@
+ */
+
+goog.require('shaka.text.Cue');
+goog.require('shaka.text.SsaTextParser');
+goog.require('shaka.util.BufferUtils');
+goog.require('shaka.util.StringUtils');
+
+describe('SsaTextParser', () => {
+  const Cue = shaka.text.Cue;
+
+  it('supports no cues', () => {
+    verifyHelper([],
+        '',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
+  it('handles a blank line at the start of the file', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 2, payload: 'Test'},
+        ],
+        '\n\n' +
+        '[Script Info]\n' +
+        'Title: Foo\n\n' +
+        '[V4+ Styles]\n' +
+        'Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, ' +
+        'OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ' +
+        'ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, ' +
+        'Alignment, MarginL, MarginR, MarginV, Encoding\n' +
+        'Style: DefaultVCD, Arial,28,&H00B4FCFC,&H00B4FCFC,&H00000008,' +
+        '&H80000008,-1,0,0,0,100,100,0.00,0.00,1,1.00,2.00,2,30,30,30,0\n\n' +
+        '[Events]\n' +
+        'Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, ' +
+        'Effect, Text\n' +
+        'Dialogue: 0,0:00:00.00,0:00:02.00,DefaultVCD, NTP,0000,0000,0000' +
+        ',,{\\pos(400,570)}Test',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
+  it('handles a blank line at the end of the file', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 2, payload: 'Test'},
+        ],
+        '[Script Info]\n' +
+        'Title: Foo\n\n' +
+        '[V4+ Styles]\n' +
+        'Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, ' +
+        'OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ' +
+        'ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, ' +
+        'Alignment, MarginL, MarginR, MarginV, Encoding\n' +
+        'Style: DefaultVCD, Arial,28,&H00B4FCFC,&H00B4FCFC,&H00000008,' +
+        '&H80000008,-1,0,0,0,100,100,0.00,0.00,1,1.00,2.00,2,30,30,30,0\n\n' +
+        '[Events]\n' +
+        'Format: Layer, Start, End, Style, Name, MarginL, MarginR, MarginV, ' +
+        'Effect, Text\n' +
+        'Dialogue: 0,0:00:00.00,0:00:02.00,DefaultVCD, NTP,0000,0000,0000' +
+        ',,{\\pos(400,570)}Test' +
+        '\n\n',
+        {periodStart: 0, segmentStart: 0, segmentEnd: 0});
+  });
+
+  it('handles no blank line at the end of the file', () => {
+    verifyHelper(
+        [
+          {startTime: 0, endTime: 2, payload: 'Test'},
+        ],
+        '[Script Info]\n' +
+        'Title: Foo\n\n' +
+        '[V4+ Styles]\n' +
+        'Format: Name, Fontname, Fontsize, PrimaryColour, SecondaryColour, ' +
+        'OutlineColour, BackColour, Bold, Italic, Underline, StrikeOut, ' +
+        'ScaleX, ScaleY, Spacing, Angle, BorderStyle, Outline, Shadow, ' +
+        'Alignment, MarginL, MarginR, MarginV, Encoding\n' +
```

Comment:
Are all these required?  It would be easier to understand if you had these tests have a minimum size.

Reply:
Sorry, I do not agree, if I delete and leave the minimum then I would no longer be testing SubStation Alpha but a new invented format. Even so if you think it is not reasonable, I change it, but I prefer that you understand the reason why I have done it like this.

Reply:
That's what I was asking: is this required by the spec.  If these are required for a valid file, it's fine to keep.  I was suggesting to create the smallest valid file we can to make it easier to understand.